rpc/auth.go

```diff
@@ -0,0 +1,41 @@
+package rpc
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/golang-jwt/jwt/v4"
+)
+
+// HeaderAuthProvider is an interface for adding JWT Bearer Tokens to HTTP/WS (on the initial upgrade)
+// requests to authenticated APIs.
+// See https://github.com/ethereum/execution-apis/blob/main/src/engine/authentication.md for details
+// about the authentication scheme.
+type HeaderAuthProvider interface {
+	// AddAuthHeader adds an up to date Authorization Bearer token field to the header
+	AddAuthHeader(header *http.Header) error
+}
+
+type JWTAuthProvider struct {
+	secret []byte
+}
+
+// NewJWTAuthProvider creates a new JWT Auth Provider.
+// The secret should not be empty.
+func NewJWTAuthProvider(jwtsecret []byte) *JWTAuthProvider {
+	return &JWTAuthProvider{secret: jwtsecret}
```

Comment:
This could check some info about the validity of the secret, i.e length - I'm not sure how the error would surface, but surfacing it here might be easier.

Reply:
Changed it to take a `[32]byte` secret instead, avoiding the length edge cases without introducing more errors to handle.

The spec says: "which designates a file containing the hex-encoded 256 bit secret key to be used for verifying/generating jwt tokens." about the cli option, but then also "If such a parameter is given, but the file cannot be read, or does not contain a hex-encoded key of at least 256 bits, the client should treat this as an error".

Internally the jwt library just hashes the key if it's longer than 256 bits, and it seems reasonable to enforce that length based on spec wording.